app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,44 @@
+class FoodsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_food, only: %i[show]
+
+  def index
+    @foods = current_user.foods.includes(:user)
+  end
+
+  def new
+    @food = Food.new
+  end
+
+  def create
+    @food = current_user.foods.build(food_params)
+    if @food.save
+      redirect_to foods_path, notice: 'Food item added successfully.'
+    else
+      render :new
+    end
+  end
+
+  def show
+    @food = Food.find(params[:id])
+    if @food.recipe_foods.empty?
+      if @food.destroy
+        redirect_to foods_path, notice: 'Food item deleted.'
+      else
+        redirect_to foods_path, alert: 'Failed to delete food item.'
+      end
+    else
+      redirect_to foods_path, alert: 'Food item is still being used in recipes'
+    end
+  end
```

Comment:
- [ ] Great progress on your implementation so far! 👏 While your code is functional, I'd like to emphasize the importance of using the appropriate action for each functionality. Specifically, I noticed that you've implemented delete logic within the show action. While this approach technically works, it's essential to consider maintainability for both yourself and future developers.

   - To enhance the clarity and maintainability of your code, I recommend moving the delete logic to the destroy action. The destroy action conventionally handles delete operations, making it easier for other developers to understand and maintain your code in the future.

Keep up the excellent work! 😊👨‍💻 Happy coding! ✅